aspnetcore/includes/blazor-security/azure-scope.md

```diff
@@ -1,14 +1,67 @@
+::: moniker range=">= aspnetcore-5.0"
+
+> [!NOTE]
+> If the app throws an unhandled exception when it receives a *401 Unauthorized* response from the API, inspecting the token error may reveal that the access token's audience isn't correct. This scenario is most likely to occur when the App ID URI of your *`Server`* app isn't `api://{SERVER API APP CLIENT ID OR CUSTOM VALUE}` but instead is in the format `https://{TENANT}.onmicrosoft.com/{SERVER API APP CLIENT ID OR CUSTOM VALUE}`. If that's the case, the default access token scope in `Program.Main` (`Program.cs`) of the *`Client`* app appears similar to the following after the solution is created from the Blazor Hosted template:
```

Comment:
The difference between these two URIs depends on if the application is configured as B2C or AD. I would clarify that and make sure that people match the type of auth they are using with the scope that is provided.

Also, maybe add a note about how the templates will take care of setting the correct scope?

Reply:
> make sure that people match the type of auth they are using with the scope that is provided.

Implies that the initial setup and configuration guidance is incorrect, but Javier couldn't guide me when we set the topic up as to what's wrong with what it says. 

Take the Azure portal aspect ... It doesn't permit me to set the App ID URI outside of the server app client ID segment. That's never been explained ... do you know why my portal won't allow me to use an `api://`-based URI?

Reply:
@jmprieur ... Can I trouble u for a few minutes to look at something? If you have time, read my remarks in the opening comment under **Scopes ... scopes ... and SCOPES!** :point_up:. I noticed that you wrote ...

[What if the App ID URI of your application is not api://{ClientID}](https://github.com/AzureAD/microsoft-identity-web/wiki/Web-APIs#what-if-the-app-id-uri-of-your-application-is-not-apiclientid)

... and that's exactly what I need to make my hosted Blazor WASM with AAD work.

You said ...

> This is the default when your register your application with the application registration portal. 

... but it seems **more restrictive** than just a default. My portal won't allow me to have an `api://`-based App ID URI at all. It defaults to the format ...

```
https://{TENANT}.onmicrosoft.com/{SERVER API APP CLIENT ID OR CUSTOM VALUE}
```

... and will only allow me to change the segment after the `https://{TENANT}.onmicrosoft.com/` part. Here's what I see ...

<img width="595" alt="Capture" src="https://user-images.githubusercontent.com/1622880/95520519-15759c00-098d-11eb-88fa-993bcc8aa989.PNG">

Do you know why some of us, myself and a number of community members, can't change the App ID URI in the portal and why it defaults to that format?

WRT to the coverage here ... we have it. :+1: I assume in the coverage that the reader will receive the `api://`-based value and won't require additional config (i.e., `Authority` setting). **_BUT_** ... it is covered in a NOTE that will steer devs in the right direction if they have a portal like mine. I'm basically handling this the same way that you did with that section that you wrote.

Reply:
@guardrex : I'd think that this is because the tenant in which you are working has a verified domain. Do you want us to take this offline to verify?

Reply:
@jmprieur ... :tada: **_BINGO!_** :tada: ... That's it. A virtual case of :beer: to you for that clarification. When I switched to my default directory, the publisher domain is verified, sooooooo ......

<img width="253" alt="Capture" src="https://user-images.githubusercontent.com/1622880/95597109-d9394e80-0a13-11eb-8a78-54de1b57227f.PNG">

<img width="304" alt="Capture1" src="https://user-images.githubusercontent.com/1622880/95597120-dcccd580-0a13-11eb-9c45-91aca974176a.PNG">

Now that I understand this, I can clarify it in the doc for our readers. I'll probably cross-link [How to: Configure an application's publisher domain](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-configure-publisher-domain) as an additional resource.